setup-kind/action.yaml

```diff
@@ -141,9 +149,11 @@ runs:
 
         # This is needed in order to support projected volumes with service account tokens.
         # See: https://kubernetes.slack.com/archives/CEKK1KTN2/p1600268272383600
+        #
+        # We don't specify the API version since Kind is permissive when patching
+        # https://kubernetes.slack.com/archives/CEKK1KTN2/p1648855849643189?thread_ts=1648855553.841399&cid=CEKK1KTN2
```

Comment:
FYI something about this breaks our downstream CI / `sigstore/scaffolding` because the Cluster issuer ends up being `https://kubernetes.default.svc.cluster.local` instead of `https://kubernetes.default.svc` as specified below.

Was there a particular reason you wanted this change?  For now I'm going to revert this piece and keep the rest.

cc @vaikas

Reply:
> Was there a particular reason you wanted this change? For now I'm going to revert this piece and keep the rest.

Without that change I couldn't change cluster domain

Reply:
Do you mean the suffix randomization stuff?

Reply:
I wonder if it's actually working now? Your default is 

https://github.com/chainguard-dev/actions/blob/91a1f88339ee6c46de750b7e22cc000d1cf20f07/setup-kind/action.yaml#L56

Reply:
It was originally copied from here: https://github.com/mattmoor/mink/blob/b9148a39b2d8bbc69ca9aaf5e89a7613c0b179d8/hack/setup-kind.sh#L151-L164

Reply:
> It was originally copied from here

Yeah and that API version doesn't work on 1.23 clusters

Reply:
FWIW the other option you could do is have one patch per config version (for kubeadm, kubelet etc) that you need, depending on the Kubernetes releases you target. Eliding the version simplifies that when you know you're using a field that has been stable across the versions you need (which ... is often the case), kind's patch runtime will accept this and just match on the GK instead of GVK.* 

We try hard to make sure that KIND is using the current version of the component configs once they're introduced into Kubernetes, so you can generally assume that if v1beta3 is the latest version in the k8s component at that k8s version, that's what kind will use.

\* There's an open issue to fix this for json6902 patches which always require GVK currently.

Reply:
If you do one patch per component config version kind will apply the matching patch(es) only. 

Currently we just ignore non-matching patches and changing that would probably break too many people, though it might be worth revisiting in a future KIND config version.